tests/pos-macros/i18228.scala

```diff
@@ -0,0 +1,13 @@
+import scala.quoted.*
+
+case class QueryMeta[T](map: Map[String, String])
+
+object QueryMeta:
+  given [T: Type]: FromExpr[QueryMeta[T]] = new FromExpr[QueryMeta[T]]:
+    def unapply(expr: Expr[QueryMeta[T]])(using q: Quotes): Option[QueryMeta[T]] =
+      import q.reflect.*
+      expr match
+        case '{ QueryMeta[t](${ map }: Map[String, String]) } =>
```

Comment:
```suggestion
        case '{ QueryMeta(${ map }: Map[String, String]) } =>
```

Shouldn't we use only either instance of `Type[T]` (variant a) or only type extractor `[t]` (variant b)?

Reply:
This one was supposed to be 

```scala
        case '{ QueryMeta[T](${ map }: Map[String, String]) } =>
```

I pushed a fixed version.